### PR TITLE
HBASE-24575: Cleanup the build directory structure

### PR DESCRIPTION
**Pull request by bharathv**
Switches the build layout to the following. Logically groups
all the build artifacts to make it easy to work with.
.
├── bin
├── CMakeCache.txt
├── CMakeFiles
├── cmake_install.cmake
├── CTestTestfile.cmake
├── dependencies
├── libs
├── Makefile
└── tests

### CODE REVIEW

**Comment by bharathv** [comment]
@joshelser @phrocker Here is another one that makes the build directory structure and easy to work with. FYI.

File: cmake/DownloadWangle.cmake [hunk 1]
```diff
@@ -20,24 +20,24 @@
...
-		URL "https://github.com/facebook/wangle/archive/v2017.09.04.00.tar.gz"
-		PATCH_COMMAND ${PATCH_FOLLY}
-		INSTALL_DIR "${BUILD_DIR}/dependencies/facebook-wangle-proj-install"
-		 CONFIGURE_COMMAND ${CMAKE_COMMAND} -DBUILD_EXAMPLES=OFF -DCMAKE_CROSSCOMPILING=ON -DBUILD_TESTS=OFF -DFOLLY_ROOT_DIR=${FOLLY_ROOT_DIR} -DCMAKE_INSTALL_PREFIX:PATH=${BUILD_DIR}/dependencies/facebook-wangle-proj-install
-        	${BUILD_DIR}/facebook-wangle-proj-prefix/src/facebook-wangle-proj/wangle # Tell CMake to use subdirectory as source.
-		)
-
-
-	set(WANGLE_ROOT_DIR "${BUILD_DIR}/dependencies/facebook-wangle-proj-install" CACHE STRING "" FORCE)
+  set(WANGLE_DOWNLOAD_DIR "${BUILD_DIR}/dependencies/facebook-wangle-proj-download")
```
Comment (phrocker) [inline cmake/DownloadWangle.cmake:23]: Ah, you've mentioned the spacing before I will take note to make sure I follow this next time. Thanks!

**Review by phrocker — APPROVED** [review]
@bharathv I think this is great!

File: cmake/DownloadWangle.cmake (hunk 1 above)
Comment (bharathv) [reply cmake/DownloadWangle.cmake:23]: No worries.

**Review by bharathv — COMMENTED** [review]
Thanks for the review @phrocker . Rebased on top of my other change that I just committed.

I think additional change we can make here is to put all the includes in the "include" dir in the build root. Helpful for the users building against this client. That can be done once we start prepping for a release.

**Pull request merged**